src/math/math_base.h

```diff
@@ -105,5 +106,67 @@ struct PointBase3D : public PointBase2D<_Type>
     _Type z;
 };
 
+template <typename _Type>
+struct LineBase2D
+{
+    LineBase2D( const PointBase2D<_Type> & _p1, const PointBase2D<_Type> & _p2 )
+        : p1( _p1 )
+        , p2( _p2 )
+    { }
+
+    LineBase2D( const PointBase2D<_Type> & _p, _Type angleRadians )
+        : p1( _p )
+        , p2( _p.x + std::cos(angleRadians), _p.y + std::sin(angleRadians) )
+    { }
+
+    bool operator == ( const LineBase2D & line ) const
+    {
+        return p1 == line.p1 && p2 == line.p2;
+    }
+
+    LineBase2D operator + ( const PointBase2D<_Type> & point )
+    {
+        return LineBase2D(p1 + point, p2 + point);
+    }
+
+    // This is translation (shift) function
+    LineBase2D & operator += ( const PointBase2D<_Type> & point )
+    {
+        *this = *this + point;
```

Comment:
Here is no mistake in code. Just thinking from performance side: such code requires 2 copy operations. If we simply do:
```cpp
p1 += point;
p2 += point;
return *this;
```
we need only 1 copy operation.